src/System.Text.Json/tests/Serialization/Object.ReadTests.cs

```diff
@@ -63,6 +106,34 @@ public static void ReadClassWithComplexObjects()
             obj.Verify();
         }
 
+        [Theory]
+        [InlineData("", " ")]
+        [InlineData("", "\t ")]
+        [InlineData("", "//Single Line Comment\r\n")]
+        [InlineData("", "/* Multi\nLine Comment */")]
+        [InlineData("", "\t\t\t\n// Both\n/* Comments */")]
+        [InlineData(" ", "")]
+        [InlineData("\t ", "")]
+        [InlineData(" \t", " \n")]
+        [InlineData("// Leading Comment\n", "")]
+        [InlineData("/* Multi\nLine\nComment */ ", "")]
+        [InlineData("/* Multi\nLine\nComment */ ", "\t// trailing comment\n ")]
+        public static void ReadComplexClassIgnoresLeadingOrTrailingTrivia(string leadingTrivia, string trailingTrivia)
+        {
+            var options = new JsonSerializerOptions();
+            options.ReadCommentHandling = JsonCommentHandling.Skip;
+
+            ClassWithComplexObjects obj = JsonSerializer.Parse<ClassWithComplexObjects>(leadingTrivia + ClassWithComplexObjects.s_json + trailingTrivia, options);
+            obj.Verify();
+
+            // Throws due to JsonDocument.TryParse not supporting Allow
+            //var options = new JsonSerializerOptions();
+            //options.ReadCommentHandling = JsonCommentHandling.Allow;
+            //
+            //obj = JsonSerializer.Parse<ClassWithComplexObjects>(leadingTrivia + ClassWithComplexObjects.s_json + trailingTrivia, options);
```

Comment:
nit: I think the most intuitive solution here would be that the serializer rejects Allow comments, which is why I think we should remove the commented out test code like this. I have filed an issue instead: https://github.com/dotnet/corefx/issues/37634